perf: switch from image-rs to fast_image_resize for better resize performance

ok-nick opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->
Switches from image-rs to fast_image_resize to provide a ~23.261% improve in performance for a 108MB file during encoding. For a 18KB file (used in the test case), improvements were seen ~3.2%. 

## Description

<!--- Describe your changes in detail -->
When calling `ModelImage::try_into`, it now resizes and crops via fast_image_resize, noticeably improving performance.

## Related Issue

<!--- This project only accepts pull requests related to open issues -->
<!--- If suggesting a new feature or change, please discuss it in an issue first -->
<!--- If fixing a bug, there should be an issue describing it with steps to reproduce -->
<!--- Please link to the issue here: -->

## Motivation and Context

<!--- Why is this change required? What problem does it solve? -->

## How Has This Been Tested?

<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->
<img width="662" alt="image" src="https://github.com/user-attachments/assets/c94d5406-756a-4dc1-9d5a-20ac04b88126" />

## Screenshots (if appropriate):
`images/ufo_240.jpg` (18KB) - `main` branch
<img width="535" alt="image" src="https://github.com/user-attachments/assets/4fbf6ef5-7867-4843-baf9-999aa68b9500" />

`images/ufo_240.jpg` (18KB) - `ok-nick/fir` branch
<img width="552" alt="image" src="https://github.com/user-attachments/assets/1d06722a-4017-4b3b-b0c1-7de2dc85d10d" />

noise png (108MB) - `main` branch
<img width="535" alt="image" src="https://github.com/user-attachments/assets/ff80a455-f197-4c18-a393-9609abf94154" />

noise png (108MB) - `ok-nick/fir` branch
<img width="549" alt="image" src="https://github.com/user-attachments/assets/b7242d65-4512-4724-9ffc-edd54a6f4d79" />

## Types of changes

<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->

- [ ] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

andrewhalle commented:
When I run `trustmark-cli` on an image I get this failure:

```
$ cargo run -p trustmark-cli -- -m ./models encode --variant P -i ../images/ufo_240.jpg -o ./encoded.jpg
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 0.12s
     Running `target/debug/trustmark -m ./models encode --variant P -i ../images/ufo_240.jpg -o ./encoded.jpg`

thread 'main' panicked at crates/trustmark-cli/src/main.rs:122:69:
called `Result::unwrap()` on an `Err` value: ImageProcessing(Shape(ShapeError/IncompatibleShape: incompatible shapes))
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

Not yet sure why the roundtrip test doesn't catch it, the roundtrip test does the same thing

andrewhalle commented:
Okay! I know what's going on. The UFO image is 240x240. My previous code was calling `resize_exact()` which was actually scaling up for the UFO image. This commit uses an image from `crop_imm()` as the target, and `crop_imm()` results in an unchanged image if the width/height is bigger than the input image. Hence, the shape mismatch when we move into array-land.

The reason the roundtrip test doesn't fail is it doesn't use the UFO image! But the benchmark does use the UFO image, which is why it's so fast, it's failing fast and the error is being ignored. So we should add a roundtrip test that uses the UFO image as well.

ok-nick commented:
Fixed the issue and added a unit test, the benchmarks are much more reasonable now (I will upload new ones shortly). The code isn't pretty but I don't think we a whole lot of options if we want to convert any image type to RGB32F. fast_image_resize provides a [`PixelComponentMapper`](https://docs.rs/fast_image_resize/latest/fast_image_resize/struct.PixelComponentMapper.html), but that isn't capable of removing the alpha channel.

I also had an error in the CLI where the encoded image was being converted to RGBA then saved as a JPEG (which doesn't support an alpha channel). I changed it to RGB, but it's interesting that this error didn't occur on main.

andrewhalle commented:
> I also had an error in the CLI where the encoded image was being converted to RGBA then saved as a JPEG (which doesn't support an alpha channel). I changed it to RGB, but it's interesting that this error didn't occur on main.

What command did you use to produce this error? Curious.

andrewhalle commented:
Nevermind, I reproduced. Guessing it has something to do with the `image` version update, perhaps [this](https://github.com/image-rs/image/blob/main/CHANGES.md#version-0251)?
